### PR TITLE
[Event Hubs] Remove InMemoryPartitionManager for Event Processor

### PR DESCRIPTION
**Pull request by ShivangiReja**
The focus of these changes is to remove the `InMemoryPartitionManager` from the core Event Hubs client library and use in samples, to provide the intended example of an approachable on-boarding experience for new developers. 

For more details: https://github.com/Azure/azure-sdk-for-net/issues/8286

### CODE REVIEW

File: sdk/eventhub/Azure.Messaging.EventHubs/tests/Diagnostics/DiagnosticsTests.cs [hunk 1]
```diff
@@ -247,7 +247,7 @@ public async Task CheckpointManagerCreatesScope()
             var context = new PartitionContext(eventHubName, "partition");
             var data = new EventData(new byte[0], sequenceNumber: 0, offset: 0);
 
-            var processor = new EventProcessorClient("cg", new InMemoryPartitionManager(), fakeConnection, null);
+            var processor = new EventProcessorClient("cg", new MockCheckPointStorage(), fakeConnection, null);
```
Comment (jsquire) [inline sdk/eventhub/Azure.Messaging.EventHubs/tests/Diagnostics/DiagnosticsTests.cs]: @kinelski -  Do we need an actual mock implementation here, or can we get by with a stub `Mock<PartitionManager>` do you think?

**Review by jsquire — COMMENTED** [review]
I left a couple of minor comments.   The only thing that I think is significant enough to delay is moving the mock store out of the samples root.

File: sdk/eventhub/Azure.Messaging.EventHubs/samples/MockCheckPointStorage.cs [hunk 2]
```diff
@@ -9,10 +9,16 @@
 namespace Azure.Messaging.EventHubs.Processor
 {
     /// <summary>
-    ///   A volatile in-memory storage service that keeps track of checkpoints and ownership.
+    /// The EventProcessor relies on a <see cref="PartitionManager" /> to store checkpoints and handle partition
```
[the remark this answers is not in the document]
Comment (ShivangiReja) [reply sdk/eventhub/Azure.Messaging.EventHubs/samples/MockCheckPointStorage.cs]: I agree...I'll move this class to the `infrastructure`  folder.

File: sdk/eventhub/Azure.Messaging.EventHubs/samples/MockCheckPointStorage.cs [hunk 3]
```diff
@@ -9,10 +9,16 @@
 namespace Azure.Messaging.EventHubs.Processor
```
[the remark this answers is not in the document]
Comment (ShivangiReja) [reply sdk/eventhub/Azure.Messaging.EventHubs/samples/MockCheckPointStorage.cs]: Oops..forgot to update the namespace :disappointed:

File: sdk/eventhub/Azure.Messaging.EventHubs/samples/MockCheckPointStorage.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (ShivangiReja) [reply sdk/eventhub/Azure.Messaging.EventHubs/samples/MockCheckPointStorage.cs]: @kinelski let me know if you've any concerns.

**Review by jsquire — APPROVED** [review]
LGTM.    We probably want to hold on Caio's thoughts before merging.

**Comment by jsquire** [comment]
@kinelski:   PING!   Would you please take a look at the comments?  There are a couple of things that would benefit form your insight.

File: sdk/eventhub/Azure.Messaging.EventHubs/tests/EventProcessorClient/MockCheckPointStorageTests.cs [hunk 4]
```diff
@@ -7,40 +7,41 @@
 using System.Threading.Tasks;
 using Azure.Messaging.EventHubs.Processor;
 using NUnit.Framework;
+using Azure.Messaging.EventHubs.Samples.Infrastructure;
```
Comment (kinelski) [inline sdk/eventhub/Azure.Messaging.EventHubs/tests/EventProcessorClient/MockCheckPointStorageTests.cs]: Alphabetical order here as well, but `System.Threading.Tasks` should be kept at the top.

**Review by kinelski — APPROVED** [review]
LGTM. Left some minor comments.

File: sdk/eventhub/Azure.Messaging.EventHubs/samples/MockCheckPointStorage.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (kinelski) [reply sdk/eventhub/Azure.Messaging.EventHubs/samples/MockCheckPointStorage.cs]: No concerns on my part.

File: sdk/eventhub/Azure.Messaging.EventHubs/tests/Diagnostics/DiagnosticsTests.cs [hunk 5]
```diff
@@ -15,6 +15,7 @@
 using Azure.Messaging.EventHubs.Processor;
 using Moq;
 using NUnit.Framework;
+using Azure.Messaging.EventHubs.Samples.Infrastructure;
```
[the remark this answers is not in the document]
Comment (kinelski) [reply sdk/eventhub/Azure.Messaging.EventHubs/tests/Diagnostics/DiagnosticsTests.cs]: Take a look [here](https://dailydotnettips.com/automatically-place-the-system-directives-first-when-sorting-usings-option-in-visual-studio/) to set it up automatically.

[the remark this answers is not in the document]
Comment (ShivangiReja) [reply sdk/eventhub/Azure.Messaging.EventHubs/tests/Diagnostics/DiagnosticsTests.cs]: Thanks Caio :)

**Pull request merged**